sdk/identity/azure-identity/src/main/java/com/azure/identity/DefaultAzureCredential.java

```diff
@@ -35,5 +40,18 @@
      */
     DefaultAzureCredential(ArrayDeque<TokenCredential> tokenCredentials) {
         super(tokenCredentials);
+        this.tokenCredentials = new ArrayList<TokenCredential>(tokenCredentials.size());
+        this.tokenCredentials.addAll(tokenCredentials);
+    }
+
+    /**
+     * Get the list of credentials sequentially used by {@link DefaultAzureCredential} to attempt authentication.
+     * The credentials in the returned list and their order may change in future versions of Identity.
+     * This API is not intended to be used in production ready code and should only be used for development purposes.
+     *
+     * @return The list of {@link TokenCredential}.
+     */
+    public List<TokenCredential> getCredentials() {
```

Comment:
Should we move this up to ChainedTokenCredential? That way users who created a CTC can continue to extract the list for further modification.

Reply:
I think not, 
as that will allow users to alter the behavior of an already instantiated credential.